identity/src/main/java/com/android/identity/internal/Util.kt

```diff
@@ -0,0 +1,2209 @@
+        } else if (items[2].majorType == MajorType.BYTE_STRING) {
+            payload = castTo(
+                ByteString::class.java, items[2]
+            ).bytes
+        } else {
+            throw IllegalArgumentException("Item 2 (payload) is not nil or byte-string")
+        }
+        return payload
+    }
+
+    /**
+     * Returns the empty collection if no x5chain is included in the structure.
+     *
+     * Throws exception if the given bytes aren't valid COSE_Sign1.
+     */
+    @JvmStatic
+    fun coseSign1GetX5Chain(
+        coseSign1: DataItem
+    ): List<X509Certificate> {
+        val ret = ArrayList<X509Certificate>()
+        require(coseSign1.majorType == MajorType.ARRAY) { "Data item is not an array" }
+        val items = castTo(
+            co.nstant.`in`.cbor.model.Array::class.java, coseSign1
+        ).dataItems
+        require(items.size >= 4) { "Expected at least four items in COSE_Sign1 array" }
+        require(items[1].majorType == MajorType.MAP) { "Item 1 (unprotected headers) is not a map" }
+        val map = items[1] as Map
+        val x5chainItem = map[UnsignedInteger(COSE_LABEL_X5CHAIN)]
+        if (x5chainItem != null) {
+            try {
+                val factory = CertificateFactory.getInstance("X.509")
+                if (x5chainItem is ByteString) {
+                    val certBais = ByteArrayInputStream(
+                        castTo<ByteString, DataItem>(
+                            ByteString::class.java, x5chainItem
+                        ).bytes
+                    )
+                    ret.add(factory.generateCertificate(certBais) as X509Certificate)
+                } else if (x5chainItem is co.nstant.`in`.cbor.model.Array) {
+                    for (certItem in castTo<co.nstant.`in`.cbor.model.Array, DataItem>(
+                        co.nstant.`in`.cbor.model.Array::class.java, x5chainItem
+                    ).dataItems) {
+                        val certBais = ByteArrayInputStream(
+                            castTo(
+                                ByteString::class.java, certItem
+                            ).bytes
+                        )
+                        ret.add(factory.generateCertificate(certBais) as X509Certificate)
+                    }
+                } else {
+                    throw IllegalArgumentException("Unexpected type for x5chain value")
+                }
+            } catch (e: CertificateException) {
+                throw IllegalArgumentException("Unexpected error", e)
+            }
+        }
+        return ret
+    }
+
+    /* Encodes an integer according to Section 2.3.5 Field-Element-to-Octet-String Conversion
+     * of SEC 1: Elliptic Curve Cryptography (https://www.secg.org/sec1-v2.pdf).
+     */
+    fun sec1EncodeFieldElementAsOctetString(
+        octetStringSize: Int,
+        fieldValue: BigInteger?
+    ): ByteArray {
+        return BigIntegers.asUnsignedByteArray(octetStringSize, fieldValue)
+    }
```

Comment:
"equals" instead of `return`

```
    fun sec1EncodeFieldElementAsOctetString(
        octetStringSize: Int,
        fieldValue: BigInteger?
    ) = BigIntegers.asUnsignedByteArray(octetStringSize, fieldValue)
```